src/os/layer/preset/presetmenubutton.js

```diff
@@ -127,6 +137,98 @@ class Controller extends MenuButtonCtrl {
     this.scope = null;
   }
 
+  /**
+   * Clone source, then update the configs and feature actions to currently selected values
+   *
+   * @param {!osx.layer.Preset} source
+   * @return {!osx.layer.Preset}
+   */
+  clone(source) {
+    const clone = /** @type {osx.layer.Preset} */ (Object.assign({}, source));
+
+    // get the current layerConfig
+    if (clone.layerId) {
+      const layer = os.MapContainer.getInstance().getLayer(clone.layerId); // TODO fix global reference
```

Comment:
MapContainer must require something in Layers that eventually winds around back to Presets... can't fix this require() until we unwind that; which is both potentially huge and better suited for the ES6 work.